### PR TITLE
Migrate plugin to lib-go

### LINKED ISSUE #2: Plugin code needs to be reviewed
Plugin code needs to be reviewed.

### PR DESCRIPTION
**Pull request by kdembler**
This PR migrates plugin to use lib-go and does general refactor.

Resolves #2

### CODE REVIEW

File: movingaverage/movingaverage.go [hunk 1]
```diff
@@ -19,350 +19,134 @@ limitations under the License.
...
-func (p *movingAverageProcessor) addBufferData(index int, data interface{}, namespace string) error {
-	if _, ok := p.movingAverageMap[namespace]; ok {
-		if index >= len(p.movingAverageMap[namespace].movingAverageBuf) {
-			return errors.New("Incorrect value of index, trying to access non-existing element of buffer")
-		}
-		p.movingAverageMap[namespace].movingAverageBuf[index] = data
-		return nil
-	} else {
-		return errors.New("Namespace is not present in the map")
+func castToFloat(data interface{}) (float64, error) {
```
Comment (katarzyna-z) [inline movingaverage/movingaverage.go]: Wouldn't  simple "assertion" work here?
If you need here only parse to float64, you can write:
```
func castToFloat(data interface{}) (float64, error) {
	val, ok := data.(float64)
	if !ok {
		val, fmt.Errof("Conversion error, data: %v", data)
	}
	return val, nil
}
```
see also [this](https://play.golang.org/p/SUrbPU1qSp)

File: movingaverage/movingaverage.go [hunk 2]
```diff
@@ -19,350 +19,134 @@ limitations under the License.
...
-
-func NewMovingaverageProcessor() *movingAverageProcessor {
-	a := make(map[string]*average)
-	return &movingAverageProcessor{
-		movingAverageMap: a,
-		movingBufLength:  10,
-	}
-}
+const (
+	Name    = "movingaverage"
```
Comment (katarzyna-z) [inline movingaverage/movingaverage.go]: Exported consts should have a comment or be unexported.

File: movingaverage/movingaverage.go (hunk 1 above)
Comment (kdembler) [inline movingaverage/movingaverage.go]: Type assertion will work only if `data` is actually a `float`, hidden under `interface{}`. If supplied with other type, say `int`, assertion will fail.  That's why I'm using `reflect` here.

Comment (katarzyna-z) [inline movingaverage/movingaverage.go]: Ok, please add comment with this information to "protect" against unwanted changes.

**Pull request merged**